apm-agent-plugins/apm-log-correlation-plugin/src/test/java/co/elastic/apm/agent/slf4j/Slf4JMdcActivationListenerTest.java

```diff
@@ -49,12 +53,25 @@
 
     private LoggingConfiguration loggingConfiguration;
     private CoreConfiguration coreConfiguration;
+    private Boolean log4jMdcWorking;
+    private ExecutorService executorService = Executors.newSingleThreadExecutor();
 
     @BeforeEach
-    void setUp() {
+    void setUp() throws Exception {
+        org.apache.log4j.MDC.put("test", true);
+        log4jMdcWorking = (Boolean) org.apache.log4j.MDC.get("test");
```

Comment:
What is this?

Reply:
Log4j thinks that Java 11 is Java 1 and disables `ThreadLocal`s